app/domain/authentication/authn_azure/authenticator.rb

```diff
@@ -31,6 +32,23 @@ def validate_azure_token
       def validate_application_identity
 
       end
+
+      def provider_uri
+        azure_authenticator_secrets["provider-uri"]
+      end
+
+      def azure_authenticator_secrets
+        @azure_authenticator_secrets ||= @fetch_authenticator_secrets.(
+          service_id: service_id,
+          conjur_account: account,
+          authenticator_name: authenticator_name,
+          required_variable_names: required_variable_names
+        )
+      end
+
+      def required_variable_names
+        @required_variable_names ||= %w(provider-uri)
```

Comment:
I see `||=` in many places in our codebase. If I understand correctly, this statement reads
 `required_variable_names || required_variable_names = %w(provider-uri)`, where `required_variable_names` is only set if `required_variable_names` is logically false (i.e. if it's nil or false). 

Would `required_variable_names` ever be given a value elsewhere? Why do we do this in place of just `required_variable_names = %w(provider-uri)` b/c our goal is to update the `required_variable_names` to be the `provider-uri`

Reply:
This is actually memoization. It sets the value if it is not set already. We use it in our code so calculations will run only once.